lxd/storage/backend_lxd.go

```diff
@@ -681,9 +681,7 @@ func (b *lxdBackend) CreateInstanceFromBackup(srcBackup backup.Info, srcData io.
 		contentType = drivers.ContentTypeBlock
 	}
 
-	// We don't know the volume's config yet as tarball hasn't been unpacked.
-	// We will apply the config as part of the post hook function returned if driver needs to.
-	vol := b.GetVolume(volType, contentType, volStorageName, nil)
+	vol := b.GetVolume(volType, contentType, volStorageName, srcBackup.Config.Volume.Config)
```

Comment:
This is a good catch, and worth having in.
But as we have to deal with backups that were created before the backup index file had the `Config` and `Volume` properties, I think we should check that these are non-nil before  trying to access them to avoid a panic when importing an older backup.

Similar to https://github.com/lxc/lxd/pull/11206/commits/42e51c49e5cd0522f168bbe1e60f028eacc930fb#diff-029fdd0e1afdacb0e66630c0d7e813772122db9a7d036060902838db315d9c4dR742

Reply:
There's already a check for this a few lines above.